subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationStep.java

```diff
@@ -66,11 +63,11 @@ public TransformationSubject transform(TransformationSubject subjectToTransform,
             Try<ImmutableList<File>> result = transformerInvoker.invoke(transformer, primaryInput, dependencies, subjectToTransform);
 
             if (result.getFailure().isPresent()) {
-                return subjectToTransform.transformationFailed(result.getFailure().get());
+                return Try.failure(result.getFailure().get());
```

Comment:
So, no getting rid of this for loop for now?

Reply:
Not for now.